src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -292,6 +300,45 @@ public void onBlockBreak(BlockBreakEvent e) {
         }
     }
 
+    @EventHandler
+    public void onExperienceReceive(PlayerExpChangeEvent e) {
+        if (e.getAmount() > 0 && Talisman.checkFor(e, SlimefunItems.TALISMAN_WISE)) {
+            e.setAmount(e.getAmount() * 2);
+        }
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent e) {
+        Player player = e.getEntity();
+        Location respawnLocation = getSafeRespawnLocation(player);
+        DamageCause dmgCause = player.getLastDamageCause().getCause();
+
+        if (dmgCause == DamageCause.VOID && Talisman.checkFor(e, SlimefunItems.TALISMAN_RESURRECTED)) {
+            SlimefunPlugin.runSync(() -> {
+                PaperLib.teleportAsync(player, respawnLocation);
+            }, 2);
+        }
+    }
+
+    private Location getSafeRespawnLocation(@Nonnull Player player) {
+        Location savedLoc = null;
+        /* Obtain Talisman here */
+
+        if (SlimefunUtils.containsSimilarItem(player.getInventory(), talisman.getItem(), true)) {
+            savedLoc = talisman.getSavedLocation(false);
+        }
+        else if (SlimefunUtils.containsSimilarItem(player.getEnderChest(), talisman.getEnderVariant(), true)) {
+            savedLoc = talisman.getSavedLocation(true);
+        }
```

Comment:
To obtain the itemstack you would need to create a method very similar to `Talisman.checkFor`.
Perhaps you could even copy it exactly and return the ItemStack rather than a boolean.
Then you could override the original `Talisman.checkFor` to just call your method and do a null check :)